### PR TITLE
graph-tool 2.80

### PR DESCRIPTION
**Pull request by count0**
This also adds support for openmp.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/g/graph-tool.rb [hunk 1]
```diff
@@ -32,6 +32,8 @@ class GraphTool < Formula
   depends_on "gmp"
   depends_on "google-sparsehash"
   depends_on "gtk+3"
+  depends_on "libomp"
+  depends_on "llvm"
   depends_on macos: :mojave # for C++17
```
Comment (chenrui333) [inline Formula/g/graph-tool.rb:37]: I think can also remove this line
```suggestion
```

File: Formula/g/graph-tool.rb [hunk 2]
```diff
@@ -134,6 +135,13 @@ def install
     # Linux build is not thread-safe.
     ENV.deparallelize unless OS.mac?
 
+    # Enable openmp
+    if OS.mac?
+      ENV.append_to_cflags "-Xpreprocessor -fopenmp"
+      ENV.append "LDFLAGS", "-L/usr/local/opt/libomp/lib -lomp"
+      ENV.append "CPPFLAGS", "-I/usr/local/opt/libomp/include"
```
Comment (chenrui333) [inline Formula/g/graph-tool.rb:142]:
```suggestion
      ENV.append "LDFLAGS", "-L#{Formula["libomp"].opt_lib} -lomp"
      ENV.append "CPPFLAGS", "-I#{Formula["libomp"].opt_include}"
```

File: Formula/g/graph-tool.rb [hunk 3]
```diff
@@ -32,7 +32,8 @@ class GraphTool < Formula
   depends_on "gmp"
   depends_on "google-sparsehash"
   depends_on "gtk+3"
-  depends_on macos: :mojave # for C++17
+  depends_on "libomp"
```
Comment (chenrui333) [inline Formula/g/graph-tool.rb:35]: this should be macos only dep?

Comment (count0) [reply Formula/g/graph-tool.rb]: Yes.

File: Formula/g/graph-tool.rb [hunk 4]
```diff
@@ -24,15 +24,15 @@ class GraphTool < Formula
...
   depends_on "pkgconf" => :build
+  depends_on "google-sparsehash" => :build
+
   depends_on "boost"
   depends_on "boost-python3"
   depends_on "cairomm@1.14"
   depends_on "cgal"
   depends_on "freetype"
   depends_on "gmp"
-  depends_on "google-sparsehash"
```
[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/g/graph-tool.rb:33]: oh, just moved it as build dep

**Comment by count0** [comment]
@BrewTestBot test this please

**Comment by count0** [comment]
@chenrui333 The tests were canceled by mistake... they were going well.

**Comment by chenrui333** [comment]
@count0 what was the issue?

**Comment by count0** [comment]
I closed the PR by mistake and the tests were cancelled.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**